### PR TITLE
CoreNavigation MARK comments

### PR DESCRIPTION
**Pull request by Udumft**
2/4 part for #3180, continuing #3364.

Regrouped types members and added MARK comments to CoreNavigation module.
This PR is purely cosmetic. If you see any logic being altered - please highlight it! It might be more comfortable to checkout the code to view how it is structured now.

### CODE REVIEW

File: Sources/MapboxCoreNavigation/BundleAdditions.swift [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import Foundation
 
 extension Bundle {
+    
+    // MARK: Accessing SDK Bundles and .Plists
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/BundleAdditions.swift:5]:
```suggestion
    // MARK: Accessing Mapbox-Specific Bundles
```

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 2]
```diff
@@ -2,87 +2,104 @@ import Foundation
 import CoreLocation
 import MapboxDirections
 
-// MARK: - RouteController
+// MARK: Parameterizing RouteController
+
+// MARK: Rerouting logic
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:7]: Marks don’t nest, but we can do something combined like “RouteController Rerouting”, then put a `MARK: -` where non-RouteController stuff begins.

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 3]
```diff
@@ -248,29 +263,33 @@ extension RouteController {
         public static let rawLocationKey: NotificationUserInfoKey = .init(rawValue: "rawLocation")
         
         /**
-         A key in the user info dictionary of a `Notification.Name.routeControllerDidFailToReroute` notification. The corresponding value is an `NSError` object indicating why `RouteController` was unable to calculate a new route.
+         A key in the user info dictionary of a `Notification.Name.currentRoadNameDidChange` notification. The corresponding value is a `NSString` object representing the current road name.
          */
-        public static let routingErrorKey: NotificationUserInfoKey = .init(rawValue: "error")
+        public static let roadNameKey: NotificationUserInfoKey = .init(rawValue: "roadName")
+        
+        // MARK: Marking Rerouting
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:270]: For consistency with the next mark:

```suggestion
        // MARK: Monitoring Rerouting
```

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 4]
```diff
@@ -339,6 +362,9 @@ extension MapboxNavigationService {
 }
 
 public extension Notification.Name {
+    
+    // MARK: Electronic Horizon Messaging
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:366]:
```suggestion
    // MARK: Electronic Horizon Notifications
```

File: Sources/MapboxCoreNavigation/CoreConstants.swift [hunk 5]
```diff
@@ -413,6 +439,9 @@ extension RoadGraph {
 }
 
 public extension Notification.Name {
+    
+    // MARK: Switching Navigation Tiles Versions
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/CoreConstants.swift:443]:
```suggestion
    // MARK: Switching Navigation Tile Versions
```

File: Sources/MapboxCoreNavigation/DispatchTimer.swift [hunk 6]
```diff
@@ -5,12 +5,28 @@ import Dispatch
  `DispatchTimer` is a general-purpose wrapper over the `DispatchSourceTimer` mechanism in GCD.
  */
 public class DispatchTimer {
+    /**
+     Describes related `DispatchTimer`'s state.
+     */
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/DispatchTimer.swift:10]: Thanks for noticing that this was undocumented.

```suggestion
    /**
     The state of a `DispatchTimer`.
     */
```

File: Sources/MapboxCoreNavigation/NavigationService.swift [hunk 7]
```diff
@@ -316,13 +199,20 @@ public class MapboxNavigationService: NSObject, NavigationService {
...
+        }
+        
+        // Reset the GPS countdown.
+        poorGPSTimer.reset()
     }
     
-    public var indexedRouteResponse: IndexedRouteResponse {
-        router.indexedRouteResponse
-    }
+    // MARK: Controlling Navigation Service
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/NavigationService.swift:215]:
```suggestion
    // MARK: Starting and Stopping Navigation
```

File: Sources/MapboxCoreNavigation/NavigationService.swift [hunk 8]
```diff
@@ -382,18 +336,75 @@ public class MapboxNavigationService: NSObject, NavigationService {
...
+     
+     - parameter routeResponse: `RouteResponse` object, containing selection of routes to follow.
+     - parameter routeIndex: The index of the route within the original `RouteResponse` object.
+     - parameter routeOptions: The route options used to get the route.
+     */
+    convenience init(routeResponse: RouteResponse, routeIndex: Int, routeOptions options: RouteOptions) {
+        self.init(routeResponse: routeResponse, routeIndex: routeIndex, routeOptions: options, directions: nil, locationSource: nil, eventsManagerType: nil)
+    }
+    
+    // MARK: Serving the Lifecycle
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/NavigationService.swift:351]: This isn’t very intuitive, but it seems like these methods can all go under “Starting and Stopping Navigation” anyways.

File: Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift [hunk 9]
```diff
@@ -68,6 +71,8 @@ public protocol NavigationServiceDelegate: AnyObject, UnimplementedLogging {
      */
     func navigationService(_ service: NavigationService, didFailToRerouteWith error: Error)
     
+    // MARK: Progressing the Route and Arriving Events
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift:74]:
```suggestion
    // MARK: Monitoring Route Progress and Updates
```

File: Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift [hunk 10]
```diff
@@ -159,6 +166,8 @@ public protocol NavigationServiceDelegate: AnyObject, UnimplementedLogging {
      */
     func navigationServiceShouldDisableBatteryMonitoring(_ service: NavigationService) -> Bool
     
+    // MARK: Simulation Messages
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/NavigationServiceDelegate.swift:169]:
```suggestion
    // MARK: Simulating a Route
```

File: Sources/MapboxCoreNavigation/PassiveLocationManager.swift [hunk 11]
```diff
@@ -96,6 +55,8 @@ open class PassiveLocationManager: NSObject {
      */
     public weak var delegate: PassiveLocationManagerDelegate?
     
+    // MARK: Controlling Location Manager
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/PassiveLocationManager.swift:58]:
```suggestion
    // MARK: Starting and Stopping the Location Manager
```

File: Sources/MapboxCoreNavigation/PassiveLocationManager.swift [hunk 12]
```diff
@@ -314,9 +264,68 @@ open class PassiveLocationManager: NSObject {
     public static func stopRecordingHistory(writingFileWith completionHandler: @escaping HistoryFileWritingCompletionHandler) {
         Navigator.shared.stopRecordingHistory(writingFileWith: completionHandler)
     }
+    
+    // MARK: Serving The Lifecycle
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/PassiveLocationManager.swift:268]: Similarly, this is basically part of starting and stopping.

File: Sources/MapboxCoreNavigation/ReplayLocationManager.swift [hunk 13]
```diff
@@ -7,15 +7,14 @@ import CoreLocation
  adjusted by interval between locations.
  */
 open class ReplayLocationManager: NavigationLocationManager {
+    
+    // MARK: Simulation Controls
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/ReplayLocationManager.swift:11]:
```suggestion
    // MARK: Specifying What to Simulate
```

File: Sources/MapboxCoreNavigation/RouteLegProgress.swift [hunk 14]
```diff
@@ -58,6 +44,25 @@ open class RouteLegProgress: Codable {
     }
 
     public var userHasArrivedAtWaypoint = false
+    
+    // MARK: Step(s) Parameters
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteLegProgress.swift:48]:
```suggestion
    // MARK: Details About the Leg’s Steps
```

File: Sources/MapboxCoreNavigation/RouteLegProgress.swift [hunk 15]
```diff
@@ -6,28 +6,14 @@ import MapboxDirections
  `RouteLegProgress` stores the user’s progress along a route leg.
  */
 open class RouteLegProgress: Codable {
+    
+    // MARK: Leg Information
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteLegProgress.swift:10]:
```suggestion
    // MARK: Details About the Leg
```

File: Sources/MapboxCoreNavigation/RouteProgress.swift [hunk 16]
```diff
@@ -10,61 +10,82 @@ open class RouteProgress: Codable {
...
-    /**
-     Returns the remaining legs left on the current route
-     */
-    public var remainingLegs: [RouteLeg] {
-        return Array(route.legs.suffix(from: legIndex + 1))
+        return newOptions
     }
-
+    
+    // MARK: Interacting with Route
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteProgress.swift:43]:
```suggestion
    // MARK: Details About the Route
```

I think we should split off the methods into a separate mark called “Updating the Route Progress” and the computed properties into a separate mark called “Route Statistics”.

File: Sources/MapboxCoreNavigation/RouteProgress.swift [hunk 17]
```diff
@@ -105,7 +126,67 @@ open class RouteProgress: Codable {
...
+        let priorCoordinates = priorStep?.shape?.coordinates.dropLast() ?? []
+        let currentShape = currentLegProgress.currentStep.shape
+        let upcomingCoordinates = upcomingStep?.shape?.coordinates.dropFirst() ?? []
+        if let currentShape = currentShape, priorCoordinates.isEmpty && upcomingCoordinates.isEmpty {
+            return currentShape
+        }
+        return LineString(priorCoordinates + (currentShape?.coordinates ?? []) + upcomingCoordinates)
+    }
+    
+    // MARK: Leg Stats
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteProgress.swift:152]:
```suggestion
    // MARK: Leg Statistics
```

File: Sources/MapboxCoreNavigation/RouteProgress.swift [hunk 18]
```diff
@@ -133,37 +205,34 @@ open class RouteProgress: Codable {
         return legIndex + 1 < route.legs.endIndex ? route.legs[legIndex + 1] : nil
     }
     
+    // MARK: Step Stats
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteProgress.swift:208]:
```suggestion
    // MARK: Step Statistics
```

File: Sources/MapboxCoreNavigation/RouteProgress.swift [hunk 19]
```diff
@@ -133,37 +205,34 @@ open class RouteProgress: Codable {
...
-        let upcomingCoordinates = upcomingStep?.shape?.coordinates.dropFirst() ?? []
-        if let currentShape = currentShape, priorCoordinates.isEmpty && upcomingCoordinates.isEmpty {
-            return currentShape
-        }
-        return LineString(priorCoordinates + (currentShape?.coordinates ?? []) + upcomingCoordinates)
+    public var upcomingStep: RouteStep? {
+        return currentLegProgress.upcomingStep ?? upcomingLeg?.steps.first
     }
     
+    // MARK: Congestion Info
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouteProgress.swift:234]: Or perhaps more broadly, “Leg Attributes”.

File: Sources/MapboxCoreNavigation/RouterDelegate.swift [hunk 20]
```diff
@@ -63,6 +66,8 @@ public protocol RouterDelegate: AnyObject, UnimplementedLogging {
      */
     func router(_ router: Router, didFailToRerouteWith error: Error)
 
+    // MARK: Progressing the Route and Arriving Events
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/RouterDelegate.swift:69]: Did you mean “processing” rather than “progressing”?

**Review by 1ec5 — COMMENTED** [review]
Here’s some copyediting feedback about the mark titles, but I agree with the reorganization. It’ll be a lot easier to navigate these files from now on.

File: Sources/MapboxCoreNavigation/PassiveLocationManager.swift (hunk 11 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/PassiveLocationManager.swift:58]: I tent to keep this topic a bit more abstract as there is also `updateLocation` method listed, which is not exactly 'stopping' the LocationManager.

File: Sources/MapboxCoreNavigation/ReplayLocationManager.swift (hunk 13 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/ReplayLocationManager.swift:11]: Under `Specifying What to Simulate` mark I would expect methods to set coordinates/routes or something like this. But here we have not only 'what' but also 'how' to simulate. So I would suggest something wider like `Specifying Simulation`

File: Sources/MapboxCoreNavigation/RouterDelegate.swift (hunk 20 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/RouterDelegate.swift:69]: I meant "progressing" as following methods are called at various stages during user progress on the route. Also wanted to align `RouterDelegate` and `NavigationServiceDelegate` sections, so I'll use `Monitoring Route Progress and Updates` here too.

File: Sources/MapboxCoreNavigation/CoreConstants.swift (hunk 2 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/CoreConstants.swift]: Tried to rework those sections a bit

**Review by 1ec5 — APPROVED** [review]

**Pull request merged**